go/vt/vtgate/semantics/info_schema.go

```diff
@@ -0,0 +1,973 @@
+	cols = map[string]query.Type{}
+	cols["TABLESPACE_NAME"] = query.Type(6165)
+	cols["ENGINE_ATTRIBUTE"] = query.Type(2078)
+	infSchema["TABLESPACES_EXTENSIONS"] = cols
+	cols = map[string]query.Type{}
+	cols["TRIGGER_CATALOG"] = query.Type(6165)
+	cols["TRIGGER_SCHEMA"] = query.Type(6165)
+	cols["TRIGGER_NAME"] = query.Type(6165)
+	cols["EVENT_MANIPULATION"] = query.Type(2074)
+	cols["EVENT_OBJECT_CATALOG"] = query.Type(6165)
+	cols["EVENT_OBJECT_SCHEMA"] = query.Type(6165)
+	cols["EVENT_OBJECT_TABLE"] = query.Type(6165)
+	cols["ACTION_ORDER"] = query.Type(776)
+	cols["ACTION_CONDITION"] = query.Type(10264)
+	cols["ACTION_STATEMENT"] = query.Type(6163)
+	cols["ACTION_ORIENTATION"] = query.Type(6165)
+	cols["ACTION_TIMING"] = query.Type(2074)
+	cols["ACTION_REFERENCE_OLD_TABLE"] = query.Type(10264)
+	cols["ACTION_REFERENCE_NEW_TABLE"] = query.Type(10264)
+	cols["ACTION_REFERENCE_OLD_ROW"] = query.Type(6165)
+	cols["ACTION_REFERENCE_NEW_ROW"] = query.Type(6165)
+	cols["CREATED"] = query.Type(2061)
+	cols["SQL_MODE"] = query.Type(2075)
+	cols["DEFINER"] = query.Type(6165)
+	cols["CHARACTER_SET_CLIENT"] = query.Type(6165)
+	cols["COLLATION_CONNECTION"] = query.Type(6165)
+	cols["DATABASE_COLLATION"] = query.Type(6165)
+	infSchema["TRIGGERS"] = cols
+	cols = map[string]query.Type{}
+	cols["USER"] = query.Type(6167)
+	cols["HOST"] = query.Type(6167)
+	cols["ATTRIBUTE"] = query.Type(6163)
+	infSchema["USER_ATTRIBUTES"] = cols
+	cols = map[string]query.Type{}
+	cols["GRANTEE"] = query.Type(6165)
+	cols["TABLE_CATALOG"] = query.Type(6165)
+	cols["PRIVILEGE_TYPE"] = query.Type(6165)
+	cols["IS_GRANTABLE"] = query.Type(6165)
+	infSchema["USER_PRIVILEGES"] = cols
+	cols = map[string]query.Type{}
+	cols["TABLE_CATALOG"] = query.Type(6165)
+	cols["TABLE_SCHEMA"] = query.Type(6165)
+	cols["TABLE_NAME"] = query.Type(6165)
+	cols["SPECIFIC_CATALOG"] = query.Type(6165)
+	cols["SPECIFIC_SCHEMA"] = query.Type(6165)
+	cols["SPECIFIC_NAME"] = query.Type(6165)
+	infSchema["VIEW_ROUTINE_USAGE"] = cols
+	cols = map[string]query.Type{}
+	cols["VIEW_CATALOG"] = query.Type(6165)
+	cols["VIEW_SCHEMA"] = query.Type(6165)
+	cols["VIEW_NAME"] = query.Type(6165)
+	cols["TABLE_CATALOG"] = query.Type(6165)
+	cols["TABLE_SCHEMA"] = query.Type(6165)
+	cols["TABLE_NAME"] = query.Type(6165)
+	infSchema["VIEW_TABLE_USAGE"] = cols
+	cols = map[string]query.Type{}
+	cols["TABLE_CATALOG"] = query.Type(6165)
+	cols["TABLE_SCHEMA"] = query.Type(6165)
+	cols["TABLE_NAME"] = query.Type(6165)
+	cols["VIEW_DEFINITION"] = query.Type(6163)
+	cols["CHECK_OPTION"] = query.Type(2074)
+	cols["IS_UPDATABLE"] = query.Type(2074)
+	cols["DEFINER"] = query.Type(6165)
+	cols["SECURITY_TYPE"] = query.Type(6165)
+	cols["CHARACTER_SET_CLIENT"] = query.Type(6165)
+	cols["COLLATION_CONNECTION"] = query.Type(6165)
+	infSchema["VIEWS"] = cols
+	return infSchema
+}
+
+type infoSchemaWithColumns struct {
+	inner          SchemaInformation
+	infoSchemaData map[string]map[string]query.Type
+}
```

Comment:
Why adding a wrapper of our `SchemaInformation` and not adding this directly to the `VSchema` interface?